scheduler-server/scheduling-dsl-compiler/src/executor.ts

```diff
@@ -1,15 +1,56 @@
 import vm from 'vm';
 import ts from 'typescript';
 import {SourceMapConsumer} from 'source-map';
-import * as schedulerApi from './libs/scheduler-edsl-fluent-api.js';
 
-export default async function executeSourceCode(opts: { source: string, filename: string, generatedCode: string }) {
-  const transpiledSource = ts.transpileModule(opts.generatedCode + "\n" + opts.source, {
-    compilerOptions: {
-      module: ts.ModuleKind.CommonJS,
+// Fill in the missing module vm types
+declare module 'vm' {
+  export class Module {
+    dependencySpecifiers: string[];
+    error: any;
+    identifier: string;
+    namespace: unknown; // GetModuleNamespace;
+    status: 'unlinked' | 'linking' | 'linked' | 'evaluating' | 'evaluated' | 'errored';
+
+    evaluate(options?: { timeout?: number; breakOnSigInt?: boolean }): Promise<undefined>;
+
+    link(
+        linker: (
+            specifier: string,
+            extra: { assert?: { [key: string]: any } },
+            referencingModule: vm.Module,
+        ) => vm.Module | Promise<vm.Module>,
+    ): void;
+  }
+
+  export class SourceTextModule extends Module {
+    public constructor(
+        code: string,
+        options?: {
+          identifier?: string;
+          cachedData?: Buffer | NodeJS.TypedArray | DataView;
+          context?: vm.Context;
+          lineOffset?: number;
+          columnOffset?: number;
+          initializeImportMeta?: {
+            meta?: any;
+            module?: vm.SourceTextModule;
+          };
+          importModuleDynamically?: (specifier: string, importMeta: any) => Promise<vm.Module>;
+        },
+    );
+
+    createCachedData(): Buffer;
+  }
+}
+
+export default async function executeSourceCode(opts: { source: string, filename: string, dslTS: string, generatedCode: string }) {
+  const compilerOptions = {
+    module: ts.ModuleKind.ES2022,
       target: ts.ScriptTarget.ESNext,
       sourceMap: true,
-    }
+  }
+  const transpiledSource = ts.transpileModule(opts.source, {
```

Comment:
This transpiling doesn't catch/produce type errors. It simply does type annotation erasure. Probably would be good to pair program bringing in https://github.com/NASA-AMMOS/aerie-ts-user-code-runner as it handles all of that

Reply:
As Dylan and I discussed in the huddle, producing useful type errors will be important to a good user experience. That said, it is out of the scope of this PR, which is focused on providing types to the UI prior to execution.